rbac/rbac/middleware.py

```diff
@@ -420,3 +420,16 @@ def process_request(self, request):  # pylint: disable=no-self-use
 
         """
         setattr(request, "_dont_enforce_csrf_checks", True)
+
+
+class ReadOnlyApiMiddleware(MiddlewareMixin):  # pylint: disable=too-few-public-methods
+    """Middleware to enable read-only on APIs when configured."""
+
+    def process_request(self, request):  # pylint: disable=no-self-use
```

Comment:
As I checked docs it looks that django style for middleware was changed 
[example](https://docs.djangoproject.com/en/4.2/topics/http/middleware/#writing-your-own-middleware).

but we would need to change it for all middlewares - so maybe this will be better to do it in separate (probably tech debt) task - so I think it is OK 👍

Reply:
Oh, interesting! Thanks for pointing that out. Yeah, I'd agree that since this pattern is currently used throughout it would be a good candidate for tech debt.

Reply:
Added https://issues.redhat.com/browse/RHCLOUD-35536

Reply:
Thanks!!!